gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/internal/ArgumentFileGenerator.java

```diff
@@ -0,0 +1,64 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.pegasus.gradle.internal;
+
+import org.gradle.api.GradleException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+public class ArgumentFileGenerator
+{
+
+  private ArgumentFileGenerator()
+  {
+    // prevent instantiation
+  }
+
+  /**
+   *  Helper method to generate an argument file, containing one input file per line.
+   *  This should be later prefixed with '@' and passed to the final argument of RestRequestBuilderGenerator.
+   *
+   * @param prefix unique value to prepend to the arg file name
+   * @param files the iterable to write to an arg file, one line per entry
+   * @param tempDir the directory to hold the file (If invoked from Gradle, use org.gradle.api.Task#getTemporaryDir()
+   * @return argument file which can be passed to a CLI or other JavaExec task
+   */
+  public static File createArgFile(String prefix, List<String> files, File tempDir)
```

Comment:
Fair enough - although I don't intend this to become the "restli standard". We need to stop passing arguments via sysprops on the CLI, and use a proper framework like jcommander. I'm OK with commons-cli too _if_ it can support the argfile syntax. Otherwise I strongly recommend jcommander.

Reply:
Ah, I see. So for now, this will hold us over, and we'll potentially have to use it if we run into the same issue for another task in the near future. The long-term vision will be to migrate all our tools to C-CLI or JCommander.

Reply:
👍